common/bin/ceylon

```diff
@@ -36,28 +36,29 @@ else
 fi
 
 # Make sure we have java installed
-if hash java 2>&-
+if ! hash java 2>&-
 then
-    true #nothing
-else
-    echo "Java not found, you must install Java in order to compiler and run Ceylon programs"
-    echo "Go to http://www.java.com/getjava/ to download the latest version of Java"
+    echo >&2 "Java not found, you must install Java in order to compile and run Ceylon programs"
+    echo >&2 "Go to http://www.java.com/getjava/ to download the latest version of Java"
+    exit 1
 fi
 
 #JAVA_DEBUG_OPTS="-Xrunjdwp:transport=dt_socket,address=8787,server=y,suspend=y"
 
 if [ "$PRESERVE_JAVA_OPTS" != "true" ]; then
     PREPEND_JAVA_OPTS="$JAVA_DEBUG_OPTS"
-    if stty size 2>/dev/null >/dev/null; then
-        CEYL_COLS=`stty size 2>/dev/null | cut -d' ' -f2`
+    if [ -n "$COLUMNS" ]; then
+        CEYL_COLS="$COLUMNS"
+    elif stty size 2>/dev/null >/dev/null; then
+        CEYL_COLS="$(stty size 2>/dev/null | cut -d' ' -f2)"
     else
-        CEYL_COLS=`tput 2>/dev/null cols`
+        CEYL_COLS="$(tput 2>/dev/null cols)"
     fi
     PREPEND_JAVA_OPTS="$PREPEND_JAVA_OPTS -Dcom.redhat.ceylon.common.tool.terminal.width=$CEYL_COLS"
     PREPEND_JAVA_OPTS="$PREPEND_JAVA_OPTS -Dcom.redhat.ceylon.common.tool.terminal.color.reset=$(tput 2>/dev/null sgr0)"
     PREPEND_JAVA_OPTS="$PREPEND_JAVA_OPTS -Dcom.redhat.ceylon.common.tool.terminal.color.green=$(tput 2>/dev/null setaf 2)"
     PREPEND_JAVA_OPTS="$PREPEND_JAVA_OPTS -Dcom.redhat.ceylon.common.tool.terminal.color.red=$(tput 2>/dev/null setaf 1)"
```

Comment:
Incorrect in that this may change the terminal state when it’s run, and not when we later print the output.

We currently assume that `tput` works by outputting escape sequences. By saving that output and re-outputting it later, we hope to achieve the same effect, without having to call `tput` in the main Java program.

And that is how it works on virtually all systems we’re going to encounter… but that’s not how it’s *required* to work. It’s possible that `tput` is instead implemented by issuing syscalls that set terminal state, and simply won’t print anything. In this case, these three commands will set the terminal state right when we run them, and then printing `com.….reset` later won’t have any effect.

Come to think of it, perhaps we should at least issue the `sgr0` call last, so *if* we encounter such a weird terminal, at least we won’t leave a red terminal behind.